doc/syntax.md

```diff
@@ -1,16 +1,27 @@
 # Syntax Extensions
 
-Lean's syntax can be extended and customized by users at every level, ranging from basic "mixfix" notations to custom elaborators.
-In fact, all builtin syntax is parsed and processed using the same mechanisms and APIs open to users.
-In this section, we will describe and explain the various extension points.
-Significant syntax extensions already builtin into Lean such as the [`do` notation](./do.md) are described in subsections.
+[Lean's syntax](lexical_structure.md) can be extended and customized
```

Comment:
Before reformatting existing text, we should agree on a standard format and ideally enforce it in editors.